codes/Geometry/TangentPointToHull.cpp

```diff
@@ -10,18 +10,15 @@ pair<int, int> get_tangent(const vector<P> &v, P p) {
     };
     const auto gao = [&](int s) {
         const auto lt = [&](int x, int y) {
-            return cmp(v[x%N]-p, v[y%N]-p) == s; };
-        bool up = lt(0, 1);
-        int l = 0, r = N;
+          if (int c = cmp(v[x%N]-p, v[y%N]-p))
+            return c == s;
+          return norm(v[x%N]-p) < norm(v[y%N]-p);
```

Comment:
I don't think this is in the original code. Is it because of some correctness issues?

Reply:
This is mis-added. Have removed them.